docs: navbar migration documentation (#3118)
contentful/forma-36: packages/components/navbar/README.mdx

| `promotions`          | In this area you can add any additional items you want accessible on the Navbar, like feedback links or other types of promotions. This area is responsive and will not be visible until the breakpoint `1200px`.                                                                                                                                                    | `1200px` and up                  | Custom element                      |
| `switcher`            | Reserved for the `Navbar.Switcher` component to display the name of the current space and environment.                                                                                                                                                                                                                                                               | Always visible                   | `Navbar.Switcher`                   |

> | `account`             | Meant for the user's account menu. Use the `Navbar.Account` compound component to build it. This area is responsive and will be visible starting from `576px`; before that, it should be made available inside the `mobileNavigation`.                                                                                                                               | `576px` and up                   | `Navbar.Account`                    |

Reviewer: Could we move this block between the `Examples` and `Props` sections? Also, some markdown isn't supported.

<img width="1198" height="1380" alt="Image" src="https://github.com/user-attachments/assets/35eb7700-c0b6-4194-8052-f262c718e0ad" />

Author: I thought it is better to have the areas described before the examples, so i know what i am looking at. But this might be where the tech-writing team would be best to have a suggestion.

Author: Maybe adding an extra section for the responsiveness would be good, as well as something for accessibility

> > [!CAUTION]
> > The Navbar component is offering a responsive behaviour. Parts of the navbar will dissapear on certain breakpoints. Find the detailed information in the table below.
> 
> The Navbar consists of multiple areas, which can be targeted via dedicated navbar props and does not accept `children`. It is responsive and shows / hides different areas on different breakpoints.
> 
> | Prop                  | Description                                                                                                                                                                                                                                                                                                                                                          | Breakpoints                      | Recommended Compound Component      |
> | --------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------- | ----------------------------------- |
> | `logo`                | Allows to overwrite the Contentful logo with a custom logo and link. This ensures a consistent and intuitive navigation experience throughout the application.                                                                                                                                                                                                       | Always visible                   | Custom element                      |
> | `mainNavigation`      | Main navigation items. Each main navigation item should use the `Navbar.Item` compound component. This area is responsive and will be visible starting from the default breakpoint `867px`. This can be overwritten in the `mobileNavigationProps` and changed to `breakpoint: 'medium'` (`1024px`).                                                                 | `867px` (default), configurable  | `Navbar.Item`                       |
> | `mobileNavigation`    | Allows navigation on smaller screen devices. Visible until the default breakpoint of `867px`. Items in this area should use `Navbar.MenuItem` and `Navbar.Submenu` compound components. The breakpoint can be changed using `mobileNavigationProps` to `small: 867px` or `medium: 1024px`. You can also set a custom label for the mobile navigation trigger button. | `<867px` (default), configurable | `Navbar.MenuItem`, `Navbar.Submenu` |
> | `secondaryNavigation` | Contains additional navigation items, like Search, Help, etc. Items in this area should use the `Navbar.Item` compound component in the icon-only variant. This area is responsive and will be visible starting from the breakpoint `576px`.                                                                                                                         | `576px` and up                   | `Navbar.Item` (icon-only)           |
> | `promotions`          | In this area you can add any additional items you want accessible on the Navbar, like feedback links or other types of promotions. This area is responsive and will not be visible until the breakpoint `1200px`.                                                                                                                                                    | `1200px` and up                  | Custom element                      |
> | `switcher`            | Reserved for the `Navbar.Switcher` component to display the name of the current space and environment.                                                                                                                                                                                                                                                               | Always visible                   | `Navbar.Switcher`                   |
> | `account`             | Meant for the user's account menu. Use the `Navbar.Account` compound component to build it. This area is responsive and will be visible starting from `576px`; before that, it should be made available inside the `mobileNavigation`.                                                                                                                               | `576px` and up                   | `Navbar.Account`                    |

Reviewer: ```suggestion
> [!CAUTION]
> The Navbar component supports responsiveness. Parts of the navbar will disappear on some breakpoints. Find the detailed information in the table below.

The Navbar consists of multiple areas, which can be targeted via dedicated navbar props and does not accept `children`. It is responsive and shows/hides different areas on different breakpoints.

| Prop                  | Description                                                                                                                                                                                                                                                                                                                                                          | Breakpoints                      | Recommended Compound Component      |
| --------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------- | ----------------------------------- |
| `logo`                | Allows to overwrite the Contentful logo with a custom logo and link. This ensures a consistent and intuitive navigation experience throughout the application.                                                                                                                                                                                                       | Always visible                   | Custom element                      |
| `mainNavigation`      | Main navigation items. Each main navigation item should use the `Navbar.Item` compound component. This area is responsive and will be visible starting from the default breakpoint `867px`. This can be overwritten in the `mobileNavigationProps` and changed to `breakpoint: 'medium'` (`1024px`).                                                                 | `867px` (default), configurable  | `Navbar.Item`                       |
| `mobileNavigation`    | Allows navigation on smaller screen devices. Visible until the default breakpoint of `867px`. Items in this area should use `Navbar.MenuItem` and `Navbar.Submenu` compound components. The breakpoint can be changed using `mobileNavigationProps` to `small: 867px` or `medium: 1024px`. You can also set a custom label for the mobile navigation trigger button. | `<867px` (default), configurable | `Navbar.MenuItem`, `Navbar.Submenu` |
| `secondaryNavigation` | Contains additional navigation items, like Search, Help, etc. Items in this area should use the `Navbar.Item` compound component in the icon-only variant. This area is responsive and will be visible starting from the breakpoint `576px`.                                                                                                                         | `576px` and up                   | `Navbar.Item` (icon-only)           |
| `promotions`          | In this area, you can add any additional items you want accessible on the Navbar, such as feedback links or other types of promotions. This area is responsive and will not be visible until the breakpoint `1200px`.                                                                                                                                                    | `1200px` and up                  | Custom element                      |
| `switcher`            | Reserved for the `Navbar.Switcher` component to display the name of the current space and environment.                                                                                                                                                                                                                                                               | Always visible                   | `Navbar.Switcher`                   |
| `account`             | Meant for the user's account menu. Use the `Navbar.Account` compound component to build it. This area is responsive and will be visible starting from `576px`; before that, it should be made available inside the `mobileNavigation`.                                                                                                                               | `576px` and up                   | `Navbar.Account`                    |
```

1. `<Navbar>`: The main wrapper component.
1. `<Navbar.Item>`: Represents a main navigation item. Use it as a child of the `<Navbar>` component or the `bottomRightItems` prop of Navbar.
1. `<Navbar.MenuItem>`: Represents a menu item used to create dropdown menus. Use it as a child of the `<Navbar.Item>` or `<Navbar.Account>` components.

> 1. `<Navbar.Submenu>`: Wraps `<Navbar.MenuItem>`, to create a nested Menu inside the `mobileNavigation`

Reviewer 2: Wraps one or more `Navbar.MenuItem` components to create a nested menu within the `mobileNavigation` area.

1. `<Navbar>`: The main wrapper component.
1. `<Navbar.Item>`: Represents a main navigation item. Use it as a child of the `<Navbar>` component or the `bottomRightItems` prop of Navbar.
1. `<Navbar.MenuItem>`: Represents a menu item used to create dropdown menus. Use it as a child of the `<Navbar.Item>` or `<Navbar.Account>` components.

> 1. `<Navbar.Submenu>`: Used as wrapper around `<Navbar.MenuItem>` to create a nested Menu inside the `mobileNavigation`

Reviewer: ```suggestion
1. `<Navbar.Submenu>`: Wraps `<Navbar.MenuItem>`, to create a nested Menu inside the `mobileNavigation`
```

```

> ### Complete Navbar including mobile navigation

Reviewer: Line 67 - typo in `environments`

<PropsTable of="NavbarItem" />

### Navbar.Submenu

> <PropsTable of="NavbarSubmenu" />

Reviewer: It isn't being rendered. Is the name matching?

<img width="971" height="298" alt="Image" src="https://github.com/user-attachments/assets/28b8e9ec-c60b-48cd-98af-0790ec61fa33" />
